datatypes/src/primitives/time_interval.rs

```diff
@@ -299,6 +299,14 @@ impl TimeInterval {
     pub fn duration_ms(&self) -> u64 {
         self.end.inner().wrapping_sub(self.start.inner()) as u64
     }
+
+    /// Merges two time intervals such that it outs the smaller min and the larger max.
+    pub fn merge(&self, other: &Self) -> TimeInterval {
```

Comment:
i think we need a more intuitive name. something like: smallest_start_to_largest_end

Reply:
https://github.com/geo-engine/geoengine/commit/09ecc23907ba2d036f2c225a1a81da635a5880e5